File: python/leetcode/LEETCODE_9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        
        if x < 0:
            return False
        
        left, right = 0, 0
        s: str = str(x)
        
        if len(s) % 2 != 0:
            left, right = len(s) // 2, len(s) // 2
        else:
            left, right = len(s) // 2 - 1, len(s) // 2
            
        while left >= 0 and right < len(s):
            
            if s[left] != s[right]:
                return False
            
            left -= 1
            right += 1
            
            
        return True

File: java/src/leetcode/LEETCODE_9.java
package leetcode;

class LEETCODE_9 {
    public boolean isPalindrome(int x) {
        
        if (x < 0)
            return false;
        
        int left, right;
        String s = Integer.toString(x);
            
        if (s.length() % 2 != 0) {
            left = s.length() / 2;
            right = s.length() / 2;
        } else {
            left = s.length() / 2 - 1;
            right = s.length() / 2;
        }
        
        while (left >= 0 && right < s.length()) {
            
            if (s.charAt(left--) != s.charAt(right++)) {
                return false;
            }
        }
        
        return true;
    }
}
